File: frontend/src/services/searchService.ts
import { api } from './api';
import type { PageResponse, SongView } from '../types';

export const searchService = {
  async search(q: string, page = 0, size = 20): Promise<PageResponse<SongView>> {
    const res = await api.get<PageResponse<SongView>>('/api/search', {
      params: { q, page, size },
    });
    return res.data;
  },
};

File: src/main/java/com/shreya/moodify/service/SearchService.java
package com.shreya.moodify.service;
import com.shreya.moodify.dto.ApiDtos.*; import org.springframework.stereotype.Service;
@Service public class SearchService { private final SongService songs; public SearchService(SongService s){songs=s;} public PageResponse<SongView> search(String q,int page,int size){return songs.search(q,page,size);} }
